src/main/scala/outwatch/dom/helpers/DomUtils.scala

```diff
@@ -1,180 +1,339 @@
+          }
+        })
+      case h: DomUnmountHook =>
+        destroyHook = createHooksSingle(destroyHook, h.trigger)
+        domUnmountHook = createHooksSingle(domUnmountHook, h.trigger)
+      case h: DomUpdateHook =>
+        postPatchHook = createHooksPair[VNodeProxy](postPatchHook, { (oldproxy, proxy) =>
+          if (proxy._id == oldproxy._id) {
+            h.trigger(proxy, proxy)
+          }
+        })
+      case h: DomPreUpdateHook =>
+        prePatchHook = createHooksPair[VNodeProxy](prePatchHook, { (oldproxy, proxy) =>
+          if (proxy._id == oldproxy._id) {
+            h.trigger(oldproxy, proxy)
+          }
+        })
+      case h: InitHook =>
+        initHook = createHooksSingle(initHook, h.trigger)
+      case h: InsertHook =>
+        insertHook = createHooksSingle(insertHook, h.trigger)
+      case h: PrePatchHook =>
+        prePatchHook = createHooksPair(prePatchHook, h.trigger)
+      case h: UpdateHook =>
+        updateHook = createHooksPair(updateHook, h.trigger)
+      case h: PostPatchHook =>
+        postPatchHook = createHooksPair(postPatchHook, h.trigger)
+      case h: DestroyHook =>
+        destroyHook = createHooksSingle(destroyHook, h.trigger)
+      case n: NextVDomModifier =>
+        val nextModifiers = assureNextModifiers()
+        nextModifiers += n.modifier
+    }
 
-private[outwatch] sealed trait ContentKind
-private[outwatch] object ContentKind {
-  case class Dynamic(observable: Observable[Modifier], initialValue: Modifier) extends ContentKind
-  case class Static(modifier: Modifier) extends ContentKind
+    new SeparatedModifiers(proxies = proxies, attrs = attrs, props = props, styles = styles, keyOption = keyOption, emitters = emitters, initHook = initHook, insertHook = insertHook, prePatchHook = prePatchHook, updateHook = updateHook, postPatchHook = postPatchHook, destroyHook = destroyHook, domUnmountHook = domUnmountHook, hasOnlyTextChildren = hasOnlyTextChildren, nextModifiers = nextModifiers)
+  }
 }
 
-// StreamableModifiers takes a list of modifiers. It constructs an Observable
-// of updates from dynamic modifiers in this list.
-private[outwatch] class StreamableModifiers(modifiers: Seq[Modifier]) {
-
-  //TODO: hidden signature of this method (we need StaticModifier as a type)
-  //handleStreamedModifier: Modifier => Either[StaticModifier, Observable[StaticModifier]]
-  private val handleStreamedModifier: Modifier => ContentKind = {
-    case ModifierStreamReceiver(modStream) =>
-      val observable = modStream.observable.switchMap[Modifier] { mod =>
-        handleStreamedModifier(mod.unsafeRunSync) match {
-          //TODO: why is startWith different and leaks a subscription? see tests with: stream.startWith(initialValue :: Nil)
-          case ContentKind.Dynamic(stream, initialValue) => Observable.concat(Observable.now(initialValue), stream)
-          case ContentKind.Static(mod) => Observable.now(mod)
-        }
-      }
+private[outwatch] class NativeModifiers(
```

Comment:
A short comment describing what `NativeModifiers` are, would be nice.

Reply:
Will do :)